### PR TITLE
[FABN-1555] Update EventListener typedef and document

### PR DESCRIPTION
**Pull request by davidkhala**
- add RegistrationOpts definition to type.js

Signed-off-by: david <david-khala@hotmail.com>

### CODE REVIEW

**Comment by davidkhala** [comment]
@hyperledger/fabric-test-toolchain-maintainers ready for review

File: fabric-ca-client/types/index.d.ts
Comment (bestbeforetoday) [inline fabric-ca-client/types/index.d.ts:10]: Do we need blank lines between every member?

File: fabric-common/lib/EventListener.js
Comment (bestbeforetoday) [inline fabric-common/lib/EventListener.js:55]: I think this might have been left as a non-JSDoc comment so it wasn't included in the JSDoc. I don't think this constructor is intended to be called by client code. Instances are obtained only as return values from EventService.registerXXXListener() functions.

If I am correct about the intent, perhaps we could have it as a JSDoc comment but include the `@private` tag?

File: fabric-common/types/index.d.ts
Comment (bestbeforetoday) [inline fabric-common/types/index.d.ts:234]: I don't think this should be included in the public API

File: fabric-common/lib/EventListener.js
Comment (bestbeforetoday) [inline fabric-common/lib/EventListener.js:106]: This looks to me like it should be private too. It's only intended to be called internally to do the invocation of the application-supplied callback function

File: fabric-common/types/index.d.ts
Comment (bestbeforetoday) [inline fabric-common/types/index.d.ts:235]: This looks to me like it shouldn't be in the public API either

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]
The actual change to the `startBlock` and `endBlock` doc and TypeScript definitions look great to me. Just couple of other bits that I think we should change so we're not adding confusion with this change.

@harrisob Go ahead and correct my comments if I've misinterpreted the intent of any of the code

File: fabric-ca-client/types/index.d.ts
Comment (davidkhala) [reply fabric-ca-client/types/index.d.ts:17]: Indeed. To avoid conflict, I will wait until #208 is merged.

File: fabric-ca-client/types/index.d.ts
Comment (davidkhala) [reply fabric-ca-client/types/index.d.ts:10]: I will check if the blank lines styling could be found in tslint.

File: fabric-common/lib/EventListener.js
Comment (davidkhala) [reply fabric-common/lib/EventListener.js:55]: add a private tag is a good option.

File: fabric-common/types/index.d.ts
Comment (davidkhala) [reply fabric-common/types/index.d.ts:234]: Or could we keep this while we do not to export this class to outside? like what pure js do.

File: fabric-common/types/index.d.ts
Comment (davidkhala) [reply fabric-common/types/index.d.ts:235]: I believe a exported interface is a good structure of it

**Comment by davidkhala** [comment]
@bestbeforetoday would you mind triggering the builder again? My local test result show green.

File: fabric-common/types/index.d.ts
Comment (bestbeforetoday) [inline fabric-common/types/index.d.ts:240]: This should not be in the public interface

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]

File: fabric-common/types/index.d.ts
Comment (davidkhala) [reply fabric-common/types/index.d.ts:240]: Agree, WIP

**Comment by davidkhala** [comment]
Before I work on rebasing with solving a simple conflict.
I have to highlight that in this change I add back the `npm callsite` which is a critical fix. 
Without this package we could not create HSMCryptoSuite with fabric-common only! (at least for current published version)

**Review by bestbeforetoday — APPROVED** [review]

**Pull request merged**
